src/orange/g4org/LogicalVolumeConverter.cc

```diff
@@ -0,0 +1,126 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2023-2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file orange/g4org/LogicalVolumeConverter.cc
+//---------------------------------------------------------------------------//
+#include "LogicalVolumeConverter.hh"
+
+#include <G4LogicalVolume.hh>
+#include <G4Material.hh>
+#include <G4MaterialCutsCouple.hh>
+#include <G4VSolid.hh>
+
+#include "corecel/Assert.hh"
+#include "corecel/io/Logger.hh"
+#include "corecel/sys/Environment.hh"
+#include "geocel/GeantGeoUtils.hh"
+
+#include "SolidConverter.hh"
+#include "Volume.hh"
+
+namespace celeritas
+{
+namespace g4org
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct with solid conversion helper.
+ */
+LogicalVolumeConverter::LogicalVolumeConverter(SolidConverter& convert_solid)
+    : convert_solid_(convert_solid)
+{
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Convert a Geant4 logical volume to a VecGeom LogicalVolume.
```

Comment:
No VecGeom involved.

```suggestion
 * Convert a Geant4 logical volume to an ORANGE LogicalVolume.
```